### DISCUSSION TITLE
Integration of the Python-Files in the DLL-File after FMU-Export

### DISCUSSION THREAD

**Discussion by Nils1305** [NTNU-IHB/PythonFMU#183] [Q&A]
Hello together,

I would like to know if it is possible to integrate the logic of the python file into the DLL file. The goal would be that I no longer need a Python file and the exported FMU and the recipient of the FMU can not see the code. 

Finally only the dll file (and the model description) will be needed to import the FMU into a co-simulator.

Many thanks in advance.

Many greetings
Nils

**Comment by markaren** [comment]
I understand what you propose, and it's probably possible, but it is not in scope for this project.

**Reply by markaren** [reply]
Come to think about it, it should/could be done as a C/C++ FMU which happens to use Python in the implementation. E.g. use https://github.com/Ecos-platform/fmu4cpp

**Reply by Nils1305** [reply]
Thank you very much for your answer! For the project (FMU4cpp) it looks like you have to program the logic of the FMU with C++. Since I really like the PythonFMU approach and it has some advantages for me, I would like to be able to continue programming the logic in Python. Logging has also worked very well and straightforward on this. 

Do you know if there is a way to compile the Python code in C++ in the build process of the FMU to have no dependencies to Python files inside the FMU file (-->Python-Free)?

Many thanks in advance.

**Reply by markaren** [reply]
From some googling, it appears that you can replace the source file with the compiled version. I.e. replace .py with .pyc inside the FMU. But I have zero knowledge here.

https://stackoverflow.com/questions/9913193/is-it-possible-to-import-a-compiled-python-file
